File: app-react/src/pages/flutter/ui/RefreshIndicator.js
import React from 'react'
// ===================================================================== public component
import ListComponent from '@cpt/list.component'
// ===================================================================== component
const data = {
	title:'刷新小部件',
	keyword:'RefreshIndicator( )',
	type:'RefreshIndicator',
	url:'RefreshIndicatorPage',
	data:[
		{ keyword:'color', type:'Color', pro:'颜色', link:'/flutter/style/Colors' },
		{ keyword:'backgroundColor', type:'Color', pro:'背景颜色', link:'/flutter/style/Colors' },
		{ keyword:'displacement', type:'double', pro:'移位距离' },
		{ keyword:'child', type:'Widget', pro:'控件' },
		{ keyword:'semanticsLabel', type:'String', pro:'语义描述标签', code:[ {path:'flutter/public/semanticsLabel'} ]},
		{ keyword:'semanticsValue ', type:'String', pro:'语义描述值'},
		{ keyword:'notificationPredicate', type:'bool Function(ScrollNotification)', pro:'', code:[ {path:'flutter/ui/RefreshIndicator/notificationPredicate'} ]},
		{ 
			keyword:'onRefresh', type:'Future<void> Function()', pro:'', 
			code:[ {path:'flutter/ui/RefreshIndicator/onRefresh'}],
			link:[ { title:'Future', url:'/flutter/style/Future' }]
		},
	]
}
export default class Index extends React.Component{ render(){ return <ListComponent data={data} {...this.props}/> } }

File: app-flutter/lib/pages/flutter/ui/RefreshIndicator.dart
import 'package:flutter_web/material.dart';
// ======================================================================================================== 全局配置
import 'package:flutter_web_study.app/config.dart';
// ======================================================================================================== Widget 控件
class  RefreshIndicatorPage extends StatefulWidget { // 有状态的控件，控件的状态、文字等属性存在变化的可能
    final String title;
    RefreshIndicatorPage({ Key key, this.title }) : super(key: key);
    @override
    _PageState createState() => _PageState(); // === createState()  => _PageState();
}

class _PageState extends State<RefreshIndicatorPage> {
    // page
    @override
    Widget build(BuildContext context) {
        return Scaffold(
            appBar: AppBar(
                title:Config.appBar(widget.title),
                centerTitle:true,
            ),
            body: Container(
                alignment: Alignment.center,
                child: SizedBox(
                    width: 100.0,
                    height: 100.0,
                    child: RefreshIndicator(
                        color: Colors.red,
                        backgroundColor: Colors.black87,
                        child: Text('1231'),
                        displacement: 20.0,
                        onRefresh: (){
                            return Future((){
                                return 0;
                            });
                        },
                        notificationPredicate: (ScrollNotification scroll){
                            return false;
                        },
                    ),
                ),
            )
        );
    }
}